src/main/scala/FutureDeviceRep.scala

```diff
@@ -0,0 +1,27 @@
+import scala.concurrent.Future
+
+class FutureDeviceRep extends IotDeviceRepository[Future] {
+  override def registerDevice(userId: Long, serialNumber: String): Future[IotDevice] = {
+    val device = IotDevice( Storage.deviceStorageById.size, userId , serialNumber)
+    Storage.deviceStorageById = Storage.deviceStorageById + (device.id -> device)
+    Storage.deviceStorageBySn = Storage.deviceStorageBySn + (device.sn -> device)
+    Storage.deviceStorageByUserId = Storage.deviceStorageByUserId.updated(userId, Storage.deviceStorageByUserId.getOrElse(userId, Seq()) :+ device)
+    Thread.sleep(1000)
```

Comment:
if you want to simulate delay in the execution than this sleep should be inside Future:
```scala
override def registerDevice(userId: Long, serialNumber: String): Future[IotDevice] = Future {
    val device = IotDevice( Storage.deviceStorageById.size, userId , serialNumber)
    Storage.deviceStorageById = Storage.deviceStorageById + (device.id -> device)
    Storage.deviceStorageBySn = Storage.deviceStorageBySn + (device.sn -> device)
    Storage.deviceStorageByUserId = Storage.deviceStorageByUserId.updated(userId, 
    Storage.deviceStorageByUserId.getOrElse(userId, Seq()) :+ device)
    Thread.sleep(1000)
    device
}
```
otherwise it blocks current thread.